mobify/vellum
# Replace $v-space and $h-space with a single $unit

Since we first switched from `$margin` to `$v-space` and `$h-space`, they have been a source of confusion:

> Can I only use v-space for vertical metrics? Same with h-space?
> What if I need the value of one of them for something that isn’t spacing? Is that weird?
> Why is the distinction between horizontal and vertical so important?

This PR introduces `$unit` to replace both v-space and h-space. Think of it as the fundamental unit for doing layout, including things like spacing and alignment. The suggested value is between 6px and 12px (in Vellum it computes to 11px, identical to the old $v-space).

Status: **Ready for review**

Reviewers: @kpeatt @jeffkamo @nastiatikk @avelinet @cole-sanderson @mlworks 
Ticket: N/A

## Changes
- Replaces the `$v-space` and `$h-space` variables with `$unit`
- Changes instances of the old variables to the new one while keeping output values the same
- <del>Ignore sourcemaps in tests</del> <ins>Sourcemaps now ignored in master</ins>

## How to test-drive this PR
1. Checkout the master branch of vellum
2. Run `grunt` (default task) to compile the css
3. Open test/index.html in a browser; don’t close the tab
4. Check out the unit-var branch and run `grunt` again to recompile
5. Open a _new_ browser window; load up a new instance of index.html
6. Do a visual comparison of the two pages and verify things look the same

## Todos
- [x] Discussed at one Design + UI Dev joint meeting 
- [x] Designer/Developer +1
- [x] Designer/Developer +1
- [x] Designer/Developer +1
